src/Archive/ArchiveEntry.cpp

```diff
@@ -113,11 +117,28 @@ string ArchiveEntry::getName(bool cut_ext)
 	// Sanitize name if it contains the \ character (possible in WAD).
 	string saname = Misc::lumpNameToFileName(name);
 
+	return saname.BeforeFirst('.');
```

Comment:
Not sure about this. It'd essentially make the dot an invalid character for lump names. I think BeforeLast would be closer to the previous behavior of turning it into a filename and cutting out the extension.